maven-artifact/src/main/java/org/apache/maven/artifact/versioning/DefaultArtifactVersion.java

```diff
@@ -204,19 +211,33 @@ public final void parseVersion( String version )
     }
 
     private static Integer getNextIntegerToken( StringTokenizer tok )
+    {
+        String s = tok.nextToken();
+        if ( ( s.length() > 1 ) && s.startsWith( "0" ) )
+        {
+            return null;
```

Comment:
This changes the semantics of the method, no NFE thrown.

Reply:
That's intended, the callers of this method has been changed as well.